system-test/storage.ts

```diff
@@ -2250,6 +2250,37 @@ describe('storage', () => {
       });
     });
 
+    it('should skip validation if file is served decompressed', done => {
+      const filename = 'logo-gzipped.png';
+      bucket
+        .upload(FILES.logo.path, {
```

Comment:
Re: style comments - the mix of promises and callbacks makes me twitch a bit. Actually the preferred way now is to use async/await, so you could make the lambda passed to it() async and then await on stuff. I think there's a promisify function we have handy for tmp?

Reply:
for unit tests, I'd be tempted to just use `mktempSync`, I don't know that we need a module for this:

https://nodejs.org/api/fs.html#fs_fs_mkdtempsync_prefix_options

I'd also consider switching to to `async`/`await`, would probably unravel this a bunch.

Reply:
Ended up using `tmp.fileSync()` since it's already there :)
Probably gonna require more boilerplate code if we'd switch to using `fs.mkdTempSync`.